examples/ordered_list_search.cpp

```diff
@@ -0,0 +1,250 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano and the Qrack contributors 2017-2019. All rights reserved.
+//
+// This example demonstrates Grover's search, applied for the purpose of finding a value in a ordered list. (This relies
+// on the IndexedLDA/IndexedADC/IndexedSBC methods of Qrack. IndexedADC and IndexedSBC can be shown to be unitary
+// operations, while IndexedLDA is unitary up to the requirement that the "value register" is set to zero before
+// applying the operation.)
+//
+// Licensed under the GNU Lesser General Public License V3.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/lgpl-3.0.en.html
+// for details.
+
+#include <iomanip> // For setw
+#include <iostream> // For cout
+
+// "qfactory.hpp" pulls in all headers needed to create any type of "Qrack::QInterface."
+#include "qfactory.hpp"
+
+using namespace Qrack;
+
+// We align our "classical" cache to 64 bit boundaries, for optimal OpenCL performance.
+#define ALIGN_SIZE 64
+unsigned char* cl_alloc(size_t ucharCount)
```

Comment:
Should we put this under Qrack::aligned_alloc() just to avoid this having to be declared in consumers?

Reply:
I think that's a good idea. It provides a convenience without sacrificing their ability to customize it, if necessary. Where do you think this should go?

Reply:
Done - I have placed it in qinterface/protected.cpp, with the other utility methods.